lib/nerves_network/eapol_manager.ex

```diff
@@ -0,0 +1,425 @@
+################################################################
+# Copyright (C) 2022 Schneider Electric                        #
+################################################################
+
+defmodule Nerves.Network.EAPoLManager do
+
+  @moduledoc """
+  A module that wraps wpa_supplicant in order to provide the EAPoL service.
+  """
+
+  use GenServer
+  alias Nerves.Network.Types
+  alias Nerves.Network.Utils
+
+  require EEx
+  require Logger
+
+  @typep context :: Types.interface_context() | :dhcp
+
+  @typedoc "State of the GenServer."
+  @type t :: %{
+    context: context,
+    ifname: Types.ifname | nil,
+    settings: Nerves.Network.setup_settings | nil,
+    dhcp_pid: GenServer.server | nil,
+    wpa_pid: GenServer.server | nil,
+    supplicant_port: GenServer.server | nil,
+  }
+
+  @type setup() :: %{
+    :ssid => String.t(),
+    :identity => String.t(),
+    :ca_cert => String.t(),
+    :client_cert => String.t(),
+    :private_key => String.t(),
+    :private_key_passwd => String.t() | nil
+  }
+
+  # The following are Nerves locations of the supplicant. If not using
+  # Nerves, these may be different.
+  @wpa_supplicant_path    "/usr/sbin/wpa_supplicant"
+  @wpa_cli_path           "/usr/sbin/wpa_cli"
+  @wpa_control_path       "/var/run/wpa_supplicant"
+  @wpa_config_file_prefix "/var/run/nerves_network_wpa_eapol.conf"
+
+  @doc false
+  @spec start_link(Types.ifname, Nerves.Network.setup_settings, GenServer.options) :: GenServer.on_start
+  def start_link(ifname, settings, opts \\ []) do
+    GenServer.start_link(__MODULE__, {ifname, settings}, opts)
+  end
+
+  def init({ifname, settings}) do
+    # Make sure that the interface is enabled or nothing will work.
+    Logger.info("EAPoLManager(#{ifname}) starting")
+    Logger.info("Register Nerves.NetworkInterface #{inspect ifname}")
+
+    # Register for nerves_network_interface events and udhcpc events
+    {:ok, _} = Registry.register(Nerves.NetworkInterface, ifname, [])
```

Comment:
consider handling != :ok answer
{:ok, pid()} | {:error, {:already_registered, pid()}}
(not sure if this can actually happen, so just a suggestion)